src/main/java/gift/auth/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,47 @@
+package gift.auth;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.Filter;
+import java.io.IOException;
+
+public class JwtAuthenticationFilter implements Filter {
+
+    private final JwtProvider jwtProvider;
+
+    public JwtAuthenticationFilter(JwtProvider jwtProvider) {
+        this.jwtProvider = jwtProvider;
+    }
+
+    @Override
+    public void doFilter(jakarta.servlet.ServletRequest request,
+                         jakarta.servlet.ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException {
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+        String authHeader = httpRequest.getHeader("Authorization");
```

Comment:
```suggestion
        String authHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
```
이렇게 쓰면 실수할 일이 더 줄어들겠군요😊

Reply:
확실히 구현하면서 이 부분에 오타를 쳐서 에러가 났던 적이 있습니다. 감사합니다.